### PR TITLE
fix: change policies API to return and take struct

### LINKED ISSUE #9171: ListCannedPolicies should return Policy struct
https://github.com/minio/minio/blob/b1a2169dcc5b4febc4edf1f82407e28b6f695475/pkg/madmin/policy-commands.go#L53

Hi, currently I'm working on an implementation that uses the `ListCannedPolicies` function, however this function is returning a map of bytes, I think it will better to return a map of a well defined Policy `struct{}` type, that way the client doesn't need to implement its own []byte parser

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
fix: change policies API to return and take struct

This allows for order guarantees in returned values
can be consumed safely by the caller to avoid any
additional parsing and validation.

## Motivation and Context
Fixes #9171

## How to test this PR?
Breaking change need for callers to avoid further parsing and validation.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [x] Documentation needed
- [ ] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

### CODE REVIEW

**Review by Alevsk — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#9135d03b-916b-47bd-a52f-a1fc1946b88a) |
<a name="9135d03b-916b-47bd-a52f-a1fc1946b88a"></a>
### 9181-156c721/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c4.minio.io:31882
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 61f744275e9f:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 8 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 7 minutes and 15 seconds
(4/15) Running aws-sdk-ruby tests ... done in 19 seconds
(5/15) Running awscli tests ... done in 2 minutes and 35 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 6 minutes and 50 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 43 seconds
(9/15) Running minio-go tests ... done in 20 minutes and 41 seconds
(10/15) Running minio-java tests ... FAILED in 8 minutes and 24 seconds
{
  "name": "minio-java",
  "function": "composeObject(String bucketName, String objectName,List<ComposeSource> composeSources, Map <String,String > headerMap, ServerSideEncryption sseTarget)",
  "args": "size: 6 MB & 6 MB",
  "duration": 11343,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code=NotImplemented, message=A header you provided implies functionality that is not implemented, bucketName=minio-java-test-1t4dmq1, objectName=minio-java-test-1638li, resource=/minio-java-test-1t4dmq1/minio-java-test-1638li, requestId=15FE771848FAB7B2, hostId=e11ae0a6-2cbc-4344-bfec-8de46efef33a)\nrequest={method=PUT, url=http://minio-c4.minio.io:31882/minio-java-test-1t4dmq1/minio-java-test-1638li?uploadId=b8ddfeb844f4c785&partNumber=1, headers=x-amz-copy-source-if-match: 07d94e2a302656d6635c516a0b425efc-2\nx-amz-copy-source: minio-java-test-1t4dmq1/minio-java-test-14abcal\nAccept-Encoding: identity\nHost: minio-c4.minio.io:31882\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20200321T235745Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200321/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-if-match;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=501, headers=Accept-Ranges: bytes\nContent-Length: 408\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO/DEVELOPMENT.GOGET\nVary: Origin\nX-Amz-Request-Id: 15FE771848FAB7B2\nX-Xss-Protection: 1; mode=block\nDate: Sat, 21 Mar 2020 23:57:45 GMT\n}\n >>> [io.minio.MinioClient.executeReq(MinioClient.java:1220), io.minio.MinioClient.execute(MinioClient.java:1082), io.minio.MinioClient.executePut(MinioClient.java:1446), io.minio.MinioClient.executePut(MinioClient.java:1468), io.minio.MinioClient.uploadPartCopy(MinioClient.java:2619), io.minio.MinioClient.composeObject(MinioClient.java:2572), FunctionalTest.composeObject_test1(FunctionalTest.java:2587), FunctionalTest.runTests(FunctionalTest.java:3396), FunctionalTest.main(FunctionalTest.java:3514)]"
}

Executed 9 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

File: cmd/iam.go [hunk 1]
```diff
@@ -469,27 +468,23 @@ func (sys *IAMSys) InfoPolicy(policyName string) ([]byte, error) {
...
 
+	policyDocsMap := make(map[string]*iampolicy.Policy, len(sys.iamPolicyDocsMap))
+
 	for k, v := range sys.iamPolicyDocsMap {
-		data, err := json.Marshal(v)
-		if err != nil {
-			return nil, err
-		}
-		policyDocsMap[k] = data
+		policyDocsMap[k] = &v
```
Comment (vadmeste) [inline cmd/iam.go]: This is not correct, &v is always the same in each iteration, what we can do is:
```
p := v
policyDocsMap[k] = &p
```

but also I think we can avoid returning a map of pointers, I am not sure why we are doing this

**Review by vadmeste — CHANGES_REQUESTED** [review]
One issue in iam.ListPolicies()

**Review by vadmeste — APPROVED** [review]
LGTM

**Comment by harshavardhana** [comment]
There is a breaking change here we need to schedule an `mc` release along with this @kannappanr so keeping it blocked for now.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#05e198b3-0dbb-4de6-ac30-2643ca9c7966) |
| mint-gateway-azure.sh | :x: [more...](#68f4eead-8968-483a-9f61-80d65ad1edad) |
<a name="05e198b3-0dbb-4de6-ac30-2643ca9c7966"></a>
### 9181-1d68dd0/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c4.minio.io:32093
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 25b5d2a188e6:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... FAILED in 4 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-dxy1edywhetwgg",
    "objectName": "51pgzb4rivu5o3so0iqrfxnnibajna"
  },
  "duration": 766,
  "error": "RequestError: send request failed\ncaused by: Put http://minio-c4.minio.io:32093/aws-sdk-go-test-dxy1edywhetwgg/51pgzb4rivu5o3so0iqrfxnnibajna?tagging=: EOF",
  "function": "testObjectTagging",
  "message": "AWS SDK Go PUTObjectTagging expected to success but got RequestError",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 15 tests successfully.
```
<a name="68f4eead-8968-483a-9f61-80d65ad1edad"></a>
### 9181-1d68dd0/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:30922
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp ba7d8215f59e:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 9 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 7 minutes and 41 seconds
(4/15) Running aws-sdk-ruby tests ... done in 26 seconds
(5/15) Running awscli tests ... done in 2 minutes and 58 seconds
(6/15) Running healthcheck tests ... done in 1 seconds
(7/15) Running mc tests ... done in 6 minutes and 28 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 50 seconds
(9/15) Running minio-go tests ... done in 20 minutes and 45 seconds
(10/15) Running minio-java tests ... FAILED in 8 minutes and 42 seconds
{
  "name": "minio-java",
  "function": "composeObject(String bucketName, String objectName,List<ComposeSource> composeSources, Map <String,String > headerMap, ServerSideEncryption sseTarget)",
  "args": "size: 6 MB & 6 MB",
  "duration": 11566,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code=NotImplemented, message=A header you provided implies functionality that is not implemented, bucketName=minio-java-test-3i57reb, objectName=minio-java-test-1eeqb10, resource=/minio-java-test-3i57reb/minio-java-test-1eeqb10, requestId=15FEC331949C8E9C, hostId=8cc1c9f0-c9c9-4835-ac8a-e0dafa4b9790)\nrequest={method=PUT, url=http://minio-dev7.minio.io:30922/minio-java-test-3i57reb/minio-java-test-1eeqb10?uploadId=88cd5b452e9911f4&partNumber=1, headers=x-amz-copy-source-if-match: c138ff7bffab4f101c7fa80fb12ee91a-2\nx-amz-copy-source: minio-java-test-3i57reb/minio-java-test-23uen1o\nAccept-Encoding: identity\nHost: minio-dev7.minio.io:30922\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20200322T231216Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200322/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-if-match;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=501, headers=Accept-Ranges: bytes\nContent-Length: 410\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO/DEVELOPMENT.GOGET\nVary: Origin\nX-Amz-Request-Id: 15FEC331949C8E9C\nX-Xss-Protection: 1; mode=block\nDate: Sun, 22 Mar 2020 23:12:16 GMT\n}\n >>> [io.minio.MinioClient.executeReq(MinioClient.java:1220), io.minio.MinioClient.execute(MinioClient.java:1082), io.minio.MinioClient.executePut(MinioClient.java:1446), io.minio.MinioClient.executePut(MinioClient.java:1468), io.minio.MinioClient.uploadPartCopy(MinioClient.java:2619), io.minio.MinioClient.composeObject(MinioClient.java:2572), FunctionalTest.composeObject_test1(FunctionalTest.java:2587), FunctionalTest.runTests(FunctionalTest.java:3396), FunctionalTest.main(FunctionalTest.java:3514)]"
}

Executed 9 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#03967d0c-4078-4bc2-bf97-c21b0670f9d2) |
<a name="03967d0c-4078-4bc2-bf97-c21b0670f9d2"></a>
### 9181-3e1af6f/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:31647
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 72574935e75a:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 8 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 7 minutes and 34 seconds
(4/15) Running aws-sdk-ruby tests ... done in 19 seconds
(5/15) Running awscli tests ... done in 3 minutes and 25 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 6 minutes and 15 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 45 seconds
(9/15) Running minio-go tests ... done in 20 minutes and 46 seconds
(10/15) Running minio-java tests ... FAILED in 8 minutes and 41 seconds
{
  "name": "minio-java",
  "function": "composeObject(String bucketName, String objectName,List<ComposeSource> composeSources, Map <String,String > headerMap, ServerSideEncryption sseTarget)",
  "args": "size: 6 MB & 6 MB",
  "duration": 11150,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code=NotImplemented, message=A header you provided implies functionality that is not implemented, bucketName=minio-java-test-ne99cc, objectName=minio-java-test-3t2quq7, resource=/minio-java-test-ne99cc/minio-java-test-3t2quq7, requestId=15FEC6CFA045DA13, hostId=f4f22235-49f2-4b40-97cd-bb2c35814edf)\nrequest={method=PUT, url=http://minio-dev7.minio.io:31647/minio-java-test-ne99cc/minio-java-test-3t2quq7?uploadId=dcaec609333bbe17&partNumber=1, headers=x-amz-copy-source-if-match: aead91c804bc3c02f904d129dc397ef5-2\nx-amz-copy-source: minio-java-test-ne99cc/minio-java-test-19i6lcn\nAccept-Encoding: identity\nHost: minio-dev7.minio.io:31647\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20200323T001833Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200323/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-if-match;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=501, headers=Accept-Ranges: bytes\nContent-Length: 408\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO/DEVELOPMENT.GOGET\nVary: Origin\nX-Amz-Request-Id: 15FEC6CFA045DA13\nX-Xss-Protection: 1; mode=block\nDate: Mon, 23 Mar 2020 00:18:34 GMT\n}\n >>> [io.minio.MinioClient.executeReq(MinioClient.java:1220), io.minio.MinioClient.execute(MinioClient.java:1082), io.minio.MinioClient.executePut(MinioClient.java:1446), io.minio.MinioClient.executePut(MinioClient.java:1468), io.minio.MinioClient.uploadPartCopy(MinioClient.java:2619), io.minio.MinioClient.composeObject(MinioClient.java:2572), FunctionalTest.composeObject_test1(FunctionalTest.java:2587), FunctionalTest.runTests(FunctionalTest.java:3396), FunctionalTest.main(FunctionalTest.java:3514)]"
}

Executed 9 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally
Error: No such image: minio/minio:9181-3e1af6f

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#867cdfde-d76b-4163-b448-db3f996e960e) |
<a name="867cdfde-d76b-4163-b448-db3f996e960e"></a>
### 9181-f104465/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c4.minio.io:30212
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp e2b46330235f:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 13 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 8 minutes and 23 seconds
(4/15) Running aws-sdk-ruby tests ... done in 19 seconds
(5/15) Running awscli tests ... done in 3 minutes and 16 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 6 minutes and 35 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 48 seconds
(9/15) Running minio-go tests ... done in 21 minutes and 26 seconds
(10/15) Running minio-java tests ... FAILED in 8 minutes and 23 seconds
{
  "name": "minio-java",
  "function": "composeObject(String bucketName, String objectName,List<ComposeSource> composeSources, Map <String,String > headerMap, ServerSideEncryption sseTarget)",
  "args": "size: 6 MB & 6 MB",
  "duration": 11153,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code=NotImplemented, message=A header you provided implies functionality that is not implemented, bucketName=minio-java-test-10ebmnf, objectName=minio-java-test-3fvop0l, resource=/minio-java-test-10ebmnf/minio-java-test-3fvop0l, requestId=15FF7F5A2616C747, hostId=23a15259-dcce-495b-9873-a87592451d68)\nrequest={method=PUT, url=http://minio-c4.minio.io:30212/minio-java-test-10ebmnf/minio-java-test-3fvop0l?uploadId=bd7663a0535ac526&partNumber=1, headers=x-amz-copy-source-if-match: 90ae912576d1b3c7be8ab6a13b525c9a-2\nx-amz-copy-source: minio-java-test-10ebmnf/minio-java-test-3nbj3hb\nAccept-Encoding: identity\nHost: minio-c4.minio.io:30212\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20200325T084019Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200325/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-if-match;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=501, headers=Accept-Ranges: bytes\nContent-Length: 410\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO/DEVELOPMENT.GOGET\nVary: Origin\nX-Amz-Request-Id: 15FF7F5A2616C747\nX-Xss-Protection: 1; mode=block\nDate: Wed, 25 Mar 2020 08:40:19 GMT\n}\n >>> [io.minio.MinioClient.executeReq(MinioClient.java:1220), io.minio.MinioClient.execute(MinioClient.java:1082), io.minio.MinioClient.executePut(MinioClient.java:1446), io.minio.MinioClient.executePut(MinioClient.java:1468), io.minio.MinioClient.uploadPartCopy(MinioClient.java:2619), io.minio.MinioClient.composeObject(MinioClient.java:2572), FunctionalTest.composeObject_test1(FunctionalTest.java:2587), FunctionalTest.runTests(FunctionalTest.java:3396), FunctionalTest.main(FunctionalTest.java:3514)]"
}

Executed 9 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Review by dvaldivia — APPROVED** [review]

**Comment by harshavardhana** [comment]
removed blocker flag, PR is ready now because we won't be making any breaking `mc` changes. In-fact the new API will be struct based any older clients will still keep working.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#d6e0678e-b2d5-45b4-8a74-4e39a1d8bb5c) |
<a name="d6e0678e-b2d5-45b4-8a74-4e39a1d8bb5c"></a>
### 9181-b42735c/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev3.minio.io:30269
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 66cb024b2e40:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 7 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 9 minutes and 5 seconds
(4/15) Running aws-sdk-ruby tests ... done in 16 seconds
(5/15) Running awscli tests ... done in 2 minutes and 31 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 6 minutes and 4 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 27 seconds
(9/15) Running minio-go tests ... done in 18 minutes and 21 seconds
(10/15) Running minio-java tests ... done in 25 minutes and 28 seconds
(11/15) Running minio-js tests ... FAILED in 4 minutes and 31 seconds
{
  "name": "minio-js",
  "function": "listObjectsV2WithMetadata(bucketName, prefix, recursive, startAfter)",
  "args": "bucketName:minio-js-test-73edb8ec-80a1-40fd-8cd1-7c1e111a7a13, recursive:true",
  "duration": 203,
  "status": "FAIL",
  "error": "TypeError: Cannot read property '0' of undefined at /mint/run/core/minio-js/node_modules/minio/dist/main/xml-parsers.js:414:42 at Array.forEach (<anonymous>) at parseListObjectsV2WithMetadata (node_modules/minio/dist/main/xml-parsers.js:409:21) at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:88:19) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:673:10) at endReadableNT (_stream_readable.js:1201:12) at processTicksAndRejections (internal/process/task_queues.js:84:21)"
}

Executed 10 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#6cf86372-4781-4998-8dfe-b1c3bf4e3e79) |
<a name="6cf86372-4781-4998-8dfe-b1c3bf4e3e79"></a>
### 9181-d4008fa/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c4.minio.io:30403
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 5895e9cc89b8:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 7 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 9 minutes and 10 seconds
(4/15) Running aws-sdk-ruby tests ... done in 16 seconds
(5/15) Running awscli tests ... FAILED in 58 seconds
{
  "name": "awscli",
  "duration": 2087,
  "function": "aws --endpoint-url http://minio-c4.minio.io:30403 s3api create-multipart-upload --bucket \nAn error occurred (BucketAlreadyOwnedByYou) when calling the CreateBucket operation: Your previous request to create the named bucket succeeded and you already own it. --key \nAn error occurred (BucketAlreadyOwnedByYou) when calling the CreateBucket operation: Your previous request to create the named bucket succeeded and you already own it.-object\n",
  "status": "FAIL",
  "error": ""
}

Executed 4 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Pull request merged**